code/datums/mapgen/CaveGenerator.dm

```diff
@@ -46,15 +48,33 @@
 	. = ..()
 	var/start_time = REALTIMEOFDAY
 	string_gen = rustg_cnoise_generate("[initial_closed_chance]", "[smoothing_iterations]", "[birth_limit]", "[death_limit]", "[world.maxx]", "[world.maxy]") //Generate the raw CA data
+	var/blacklist = typecacheof(blacklisted_turf_types)
 
 	for(var/i in turfs) //Go through all the turfs and generate them
 		var/turf/gen_turf = i
 
+		if(blacklist && blacklist[gen_turf.type])
```

Comment:
```suggestion
		if(blacklist[gen_turf.type])
```
I'm 99% sure it's not going to runtime

Reply:
and if it does you should use || anyway

Reply:
If a cave generator doesn't define a `blacklisted_turf_types`, then `blacklist = typecacheof(null)`, and as far as I can tell when `typecacheof` receives something that is neither a path nor a list, it returns nothing implicitly.

If I follow everything correctly, `blacklist` is thus null for any cave generator without turf type blacklists. For those cave generators, I think we want to keep previous behavior - using a `||` here would mean without a blacklist it would ignore all turfs.

Reply:
It will runtime
![image](https://user-images.githubusercontent.com/26465327/131485632-bc869bfb-bb67-4b03-9573-0f21e5b7b7fb.png)
The thing that doesn't runtime is if the index is null.